packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -101,16 +121,60 @@ export default function MongoNav({
   admin = false,
   constructOrganizationURL: constructOrganizationURLProp,
   constructProjectURL: constructProjectURLProp,
+  onError = () => {},
+  onSuccess = () => {},
 }: MongoNavInterface) {
-  let data;
+  const [data, setData] = React.useState<DataInterface | undefined>(undefined);
+
+  const accountHost = hosts?.account ?? `https://account.mongodb.com`;
+  const cloudHost = hosts?.cloud ?? `https://cloud.mongodb.com`;
+  const universityHost = hosts?.university ?? `https://university.mongodb.com`;
+  const supportHost = hosts?.support ?? `https://support.mongodb.com`;
 
-  if (mode === Mode.Dev) {
-    data = devModeData;
+  function getProductionData() {
```

Comment:
We only use this function in one place. You might consider returning your fetch call from this function, and consuming the returned promise in your useEffect instead.

```js
const endpointUri = `${cloudHost}/user/shared`
const getProductionData = () => fetch(endpointUri, {
    credentials: 'include',
    method: 'GET',
})
```
```js
 useEffect(() => {
    if (mode === Mode.Dev) {
      getFixtureData().then(data => setData(data as DataInterface));
    } else {
      getProductionData()
        .then(response => response.json())
        .then(data => {
          setData(data);
          onSuccess?.(data);
        })
        .catch(error => {
          onError?.(error);
          console.error(error);
        });
      }
  }, []);
```

Reply:
👍